refinery/user_files_manager/views.py

```diff
@@ -0,0 +1,21 @@
+import logging
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from data_set_manager.utils import (search_solr, format_solr_response,
+                                    generate_solr_params_for_user)
+
+
+logger = logging.getLogger(__name__)
+
+
+class UserFiles(APIView):
+    def get(self, request):
+        params = request.query_params
+
+        solr_params = generate_solr_params_for_user(params, user=request.user)
```

Comment:
DRF handles only a few exceptions. You have to handle all other ones inside the views and probably return a more useful response than a 500 error (e.g., a descriptive JSON-encoded error message).
http://www.django-rest-framework.org/api-guide/exceptions/